### PR TITLE
Added origin & auxiliary web3 to config

### PR DESCRIPTION
**Pull request by puneet-khushwani-eth**
Have also added a public variable `auxiliaryChainId` in `FacilitatorConfig`

### CODE REVIEW

File: src/Account.ts [hunk 1]
```diff
@@ -43,9 +43,16 @@ export default class Account {
    * @param password The password required to unlock the account.
    * @returns `true` if its unlocked otherwise false.
    */
-  // public unlock(web3: Web3, password: string): boolean {
-  //   // Unlocking the account and adding it to the local web3 instance so that everything is signed
-  //   // locally when using web3.eth.send
-  //   return false;
-  // }
+  public unlock(web3: Web3, password: string): boolean {
+    // Unlocking the account and adding it to the local web3 instance so that everything is signed
```
Comment (abhayks1) [inline src/Account.ts]: Please rebase with master. Code has been merged.

File: src/Config.ts [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
 import Directory from './Directory';
-import InvalidFacilitatorConfigException from './Exception';
+import InvalidFacilitatorConfigException, { WorkerPasswordNotFoundException } from './Exception';
 import * as schema from './Config/FacilitatorConfig.schema.json';
 import Utils from './Utils';
+import Account from './Account';
 
 // Database password key to read from env.
 const ENV_DB_PASSWORD = 'MOSAIC_FACILITATOR_DB_PASSWORD';
+export const ENV_WORKER_PASSWORD_PREFIX = 'MOSAIC_ADDRESS_PASSW_';
```
Comment (abhayks1) [inline src/Config.ts]: Do we need to export this const?

File: src/Config.ts [hunk 3]
```diff
@@ -71,9 +98,22 @@ export class FacilitatorConfig {
...
    */
   private constructor(config: any) {
+    this.originChainId = config.originChainId || '';
+    this.auxiliaryChainId = '';
     this.database = config.database || new DBConfig();
-    this.chains = config.chains || {};
+    this.chains = {};
     this.encryptedAccounts = config.encryptedAccounts || {};
+    const chains = config.chains || {};
+    for (const identifier in chains) {
```
Comment (abhayks1) [inline src/Config.ts]: For loop will give linter warnings.

File: src/Config.ts [hunk 4]
```diff
@@ -71,9 +98,22 @@ export class FacilitatorConfig {
...
+      this.chains[identifier] = new Chain(
+        chains[identifier].rpc,
```
Comment (abhayks1) [inline src/Config.ts]: Should this logic go to a private method? Constructor is becoming clumsy here.

File: src/Config.ts [hunk 5]
```diff
@@ -149,16 +202,85 @@ export class Config {
...
+   */
+  public get auxiliaryWeb3(): any {
+    if (this._auxiliaryWeb3) {
+      return this._auxiliaryWeb3;
+    }
+    const auxiliaryChain = this.facilitator.chains[this.facilitator.auxiliaryChainId];
+    if (!auxiliaryChain.password) {
+      throw new WorkerPasswordNotFoundException(`password not found for ${auxiliaryChain.worker}`);
+    }
+    const account = new Account(auxiliaryChain.worker, this.facilitator.encryptedAccounts[auxiliaryChain.rpc]);
```
Comment (abhayks1) [inline src/Config.ts]: There are code duplication in originWeb3 and auxiliaryWeb3 methods. We can refactor the common pieces to a private method.

File: src/Config.ts [hunk 6]
```diff
@@ -149,16 +202,85 @@ export class Config {
...
+    if (this._originWeb3) {
+      return this._originWeb3;
+    }
+    const originChain = this.facilitator.chains[this.facilitator.originChainId];
+    if (!originChain.password) {
+      throw new WorkerPasswordNotFoundException(`password not found for ${originChain.worker}`);
+    }
+    const account = new Account(originChain.worker, this.facilitator.encryptedAccounts[originChain.rpc]);
+    this._originWeb3 = new Web3(originChain.rpc);
+    account.unlock(this._originWeb3, originChain.password);
```
Comment (abhayks1) [inline src/Config.ts]: Shouldn't account.unlock be async method?

File: test/Config/getWeb3Instance.test.ts [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+import { Config, ENV_WORKER_PASSWORD_PREFIX } from '../../src/Config';
+import SpyAssert from '../utils/SpyAssert';
+
+import Account from '../../src/Account';
+
+const Web3 = require('web3');
+
+const sinon = require('sinon');
+
+describe('Config.getWeb3Instance()', () => {
```
Comment (abhayks1) [inline test/Config/getWeb3Instance.test.ts]: Do we have method/property getWeb3Instance  in Config😱

File: src/Account.ts (hunk 1 above)
Comment (puneet-khushwani-eth) [reply src/Account.ts]: Done.

File: src/Config.ts (hunk 2 above)
Comment (puneet-khushwani-eth) [reply src/Config.ts]: We use this variable in some of our test cases thus added export here.

File: src/Config.ts (hunk 6 above)
Comment (puneet-khushwani-eth) [reply src/Config.ts]: I guess no as it calls `web3.eth.accounts.decrypt` & `web3.eth.accounts.wallet.add` which are sync operations and do not return a Promise or allow callbacks.

File: src/Config.ts [hunk 8]
```diff
@@ -71,9 +98,31 @@ export class FacilitatorConfig {
...
     this.encryptedAccounts = config.encryptedAccounts || {};
+    this.assignDerivedParams = this.assignDerivedParams.bind(this);
+    this.assignDerivedParams(config);
+  }
+
+  /**
+   * Assigns derived parameters.
+   * @param config
+   */
+  private assignDerivedParams(config: any) {
```
[the remark this answers is not in the document]
Comment (puneet-khushwani-eth) [reply src/Config.ts]: This also sets `auxiliaryChain` thus set a generic name. But open to better alternatives.

**Pull request merged**